packages/catalog-realm/fields/geo-search-point.gts

```diff
@@ -0,0 +1,268 @@
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { debounce } from 'lodash';
+import { BoxelInput } from '@cardstack/boxel-ui/components';
+import MapIcon from '@cardstack/boxel-icons/map';
+import StringField from 'https://cardstack.com/base/string';
+import {
+  Component,
+  contains,
+  field,
+} from 'https://cardstack.com/base/card-api';
+import { GeoPointField } from './geo-point';
+import { MapRender, type Coordinates } from '../components/map-render';
+
+class AtomTemplate extends Component<typeof GeoSearchPointField> {
+  get searchAddressValue() {
+    return this.args.model?.searchKey ?? 'Invalid Address';
+  }
+
+  get latValue() {
+    return this.args.model?.lat ?? 'N/A';
+  }
+
+  get lonValue() {
+    return this.args.model?.lon ?? 'N/A';
+  }
+
+  <template>
+    <div class='coordinates-section'>
+      <MapIcon class='map-icon' />
+      <span
+        class='coordinates-section-info-title'
+      >{{this.searchAddressValue}}</span>
+    </div>
+
+    <style scoped>
+      .coordinates-section {
+        display: inline-flex;
+        align-items: center;
+        gap: var(--boxel-sp-xxs);
+        flex-shrink: 0;
+      }
+
+      .coordinates-section-info-title {
+        line-height: normal;
+      }
+
+      .map-icon {
+        flex-shrink: 0;
+        color: var(--map-icon-color, var(--boxel-red));
+        width: var(--map-icon-width, 16px);
+        height: var(--map-icon-height, 16px);
+      }
+    </style>
+  </template>
+}
+
+class EditTemplate extends Component<typeof GeoSearchPointField> {
+  get searchAddressValue() {
+    return this.args.model.searchKey;
+  }
+
+  get latValue() {
+    return this.args.model?.lat ?? 'N/A';
+  }
+
+  get lonValue() {
+    return this.args.model?.lon ?? 'N/A';
+  }
+
+  private geocodeAddressInternal = task(
+    async (address: string, latField: 'lat', lonField: 'lon') => {
+      if (!address) {
+        if (this.args.model) {
+          this.args.model[latField] = undefined;
+          this.args.model[lonField] = undefined;
+        }
+        return;
+      }
+
+      try {
+        const response = await fetch(
+          `https://nominatim.openstreetmap.org/search?format=json&q=${encodeURIComponent(
+            address,
+          )}&limit=1`,
+        );
+        const data = await response.json();
+
+        if (data && data.length > 0) {
+          if (this.args.model) {
+            this.args.model[latField] = parseFloat(data[0].lat);
+            this.args.model[lonField] = parseFloat(data[0].lon);
+          }
+        }
+      } catch (error) {
+        console.error('Error geocoding address:', error);
+      }
+    },
+  );
+
+  @action
+  geocodeAddress() {
+    this.geocodeAddressInternal.perform(
+      this.args.model.searchKey || '',
+      'lat',
+      'lon',
```

Comment:
why is this such a weird api. You pass in the data right. This should be coordinate le